Use mocha for cert/redirect tests and add content-type tests

foolip opened the pull request:
Rename the directory to test to match the mocha convention, even though
the package.json is inside the directory and the test path (*.js) is
given explicitly there.

foolip commented:
This can be seen working in https://travis-ci.org/whatwg/misc-server/builds/294578969, which worked before I dropped aa66797 from the branch.

domenic commented on .travis.yml:
```diff
@@ -3,5 +3,9 @@ branches:
+before_install:
+  - cd test
+install:
+  - npm install
```
This shouldn't be necessary; it's implied by `node_js`

domenic commented on .travis.yml:
```diff
@@ -3,5 +3,9 @@ branches:
+  - npm install
 script:
-  - ./run-tests.sh
+  - npm test
```
Also unnecessary

foolip replied:
Both dropped.

domenic commented on test/certs.js:
```diff
@@ -63,35 +63,17 @@ function getCertificate(domain) {
-    try {
-      const cert = await request;
+  for (const domain of DOMAINS) {
+    it(domain, async function() {
```
Nit: I like using `specify` here instead of `it`; they are equivalent but since you're not doing `it("should do X", ...)` it reads better.

foolip replied:
Good, I didn't like `it` or understand what it meant, but didn't know there was an option.

domenic commented on test/content-type.js:
```diff
@@ -0,0 +1,157 @@
+  ['https://whatwg.org/demos/repeat-02/results.xml', 'application/xml'],
+];
+
+const WITHOUT_EXTENSION_TESTS = [
```
Is this meant to be comprehensive? Seems a pain to keep this in sync every time we add a file....

foolip replied:
It's all extensionless files I could find currently, but no, I don't think we should keep updating this, I think we should stop adding such files, or at least only use it for HTML for pretty URLs.

domenic commented on test/content-type.js:
```diff
@@ -0,0 +1,157 @@
+  ['https://whatwg.org/specs/web-forms/2004-12-10-call-for-comments/sample-multiple-editable-ui', 'image/png'],
+  ['https://whatwg.org/specs/web-forms/2004-12-10-call-for-comments/xforms-implementation-diagram', 'image/png'],
+  // much more of the same...
+  /*
```
Why keep this commented-out block?

foolip replied:
I wanted to make the tests run a bit faster by skipping some bits. And easier testing of everything when doing the nginx setup. But between deleting and including, I think I'll just run all of the tests, and maybe we can make this smaller later using http://nginx.org/en/docs/http/ngx_http_core_module.html#try_files.

foolip commented:
OK, all fixed I think, added some more things I found in https://github.com/whatwg/misc-server/issues/7. https://travis-ci.org/whatwg/misc-server/builds/294777142 passed before I reverted the .travis.yml change.
